app/partials/settings/addresses.pug

```diff
@@ -30,17 +30,21 @@
     .type-lbl.em-500(translate="UNUSED_ADDRESSES")
     p.alt-font.f-16.em-300.mvm.pbm(translate="UNUSED_ADDRESSES_COPY")
     table.accounts-container.addresses.responsive.width-100
-      tr.width-100.pointer(ng-click="createAddress()")
+      tr.width-100.pointer(ng-click="createAddress()" ng-disabled="createAddressInProgress")
         td.col-md-6
           i.ti-plus.blue.mrm
           span(translate="ADD_NEXT_ADDRESS")
         td.col-md-6
+          img(ng-show="createAddressInProgress" src="img/spinner.gif")
       tr(
         account="account"
-        ng-repeat="address in paymentRequests | filter:searchText | orderBy:'-index' track by address.index"
+        ng-repeat="address in addresses | filter:presentFilter | filter:searchText | orderBy:'-'"
         hd-address="address"
         search-text="searchText"
-        remove="removeAddressLabel(address.index, paymentRequests.indexOf(address))")
+        on-remove-label="removeAddressLabel"
```

Comment:
Could have been, especially with all the branches involved in the current release.